src/Engines/DatabaseEngine.php

```diff
@@ -8,10 +8,10 @@
 use Laravel\Scout\Attributes\SearchUsingFullText;
 use Laravel\Scout\Attributes\SearchUsingPrefix;
 use Laravel\Scout\Builder;
-use Laravel\Scout\Contracts\PaginatesEloquentModels;
+use Laravel\Scout\Contracts\PaginatesEloquentModelsUsingDatabase;
 use ReflectionMethod;
 
-class DatabaseEngine extends Engine implements PaginatesEloquentModels
+class DatabaseEngine extends Engine implements PaginatesEloquentModelsUsingDatabase
```

Comment:
Don't we need to use the both interfaces here? Technically this is still a breaking change for people relying on the previous contract to be implemented.